tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -0,0 +1,254 @@
+        Platform.environment['APP_STORE_CONNECT_ISSUER_ID'];
+    final keyIdentifier = argResults![keyIdOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_KEY_IDENTIFIER'];
+    final privateKey = argResults![privateKeyOptionName] as String? ??
+        Platform.environment['APP_STORE_CONNECT_PRIVATE_KEY'];
+
+    final isStable = stage == 'stable';
+    await runProcess(
+      'app-store-connect',
+      [
+        'publish',
+        '--path',
+        'build/ios/ipa/*.ipa',
+        '--release-type',
+        'AFTER_APPROVAL',
+        if (whatsNew != null) ...[
+          '--whats-new',
+          whatsNew,
+        ],
+        if (isStable) ...[
+          '--app-store',
+        ] else ...[
+          '--beta-group',
+          stage,
+          '--testflight',
+        ],
+        if (issuerId != null) ...[
+          '--issuer-id',
+          issuerId,
+        ],
+        if (keyIdentifier != null) ...[
+          '--key-id',
+          keyIdentifier,
+        ],
+        if (privateKey != null) ...[
+          '--private-key',
+          privateKey,
+        ],
+      ],
+      workingDirectory: _repo.sharezoneFlutterApp.location.path,
+    );
+  }
+
+  Future<int> _getNextBuildNumber() async {
+    final latestBuildNumber = await _getLatestBuildNumberFromGooglePlay();
+    final nextBuildNumber = latestBuildNumber + 1;
+    print('Next build number: $nextBuildNumber');
+    return nextBuildNumber;
+  }
+
+  /// Returns the latest build number from Google Play across all tracks.
+  Future<int> _getLatestBuildNumberFromGooglePlay() async {
+    try {
+      // From https://appstoreconnect.apple.com/apps/1434868489/
+      const appId = 1434868489;
+
+      final issuerId = argResults![issuerIdOptionName] as String? ??
+          Platform.environment['APP_STORE_CONNECT_ISSUER_ID'];
+      final keyIdentifier = argResults![keyIdOptionName] as String? ??
+          Platform.environment['APP_STORE_CONNECT_KEY_IDENTIFIER'];
+      final privateKey = argResults![privateKeyOptionName] as String? ??
+          Platform.environment['APP_STORE_CONNECT_PRIVATE_KEY'];
+
+      final result = await runProcess(
+        'app-store-connect',
+        [
+          'get-latest-build-number',
+          '$appId',
+          '--platform',
+          'IOS',
+          if (issuerId != null) ...[
+            '--issuer-id',
+            issuerId,
+          ],
+          if (keyIdentifier != null) ...[
+            '--key-id',
+            keyIdentifier,
+          ],
+          if (privateKey != null) ...[
+            '--private-key',
+            privateKey,
+          ],
+        ],
+        // Using the app location as working direcorty because the default
+        // location for the App Store Connect private key is
+        // app/private_keys/AuthKey_{keyIdentifier}.p8.
+        workingDirectory: _repo.sharezoneFlutterApp.location.path,
+      );
+      return int.parse(result.stdout);
```

Comment:
Since this is `runProcess` instead of `runProcessSucessfullyOrThrow` you haven't actually handled an error case.  
I think you should replace it with `runProcessSucessfullyOrThrow`. For this you will probably have to change the `runProcessSucessfullyOrThrow` to return a `ProcessResult`.